azurerm/internal/services/containers/resource_arm_kubernetes_cluster.go

```diff
@@ -1399,7 +1453,79 @@ func expandKubernetesClusterNetworkProfile(input []interface{}) *containerservic
 		networkProfile.ServiceCidr = utils.String(serviceCidr)
 	}
 
-	return &networkProfile
+	return &networkProfile, nil
+}
+
+func expandLoadBalancerProfile(d []interface{}, loadBalancerType string) (*containerservice.ManagedClusterLoadBalancerProfile, error) {
+	if len(d) == 0 {
+		return nil, nil
+	}
+
+	if strings.ToLower(loadBalancerType) != "standard" {
+		return nil, fmt.Errorf("Only load balancer SKU 'Standard' supports load balancer profiles. Provided load balancer type: %s", loadBalancerType)
+	}
+
+	config := d[0].(map[string]interface{})
+
+	var managedOutboundIps *containerservice.ManagedClusterLoadBalancerProfileManagedOutboundIPs
+	var outboundIpPrefixes *containerservice.ManagedClusterLoadBalancerProfileOutboundIPPrefixes
+	var outboundIps *containerservice.ManagedClusterLoadBalancerProfileOutboundIPs
+
+	if ipCount := config["managed_outbound_ips"]; ipCount != nil {
+		if c := int32(ipCount.(int)); c > 0 {
+			managedOutboundIps = &containerservice.ManagedClusterLoadBalancerProfileManagedOutboundIPs{Count: &c}
+		}
+	}
+
+	if ipPrefixes := idsToResourceReferences(config["outbound_ip_prefixes_ids"]); ipPrefixes != nil {
+		outboundIpPrefixes = &containerservice.ManagedClusterLoadBalancerProfileOutboundIPPrefixes{PublicIPPrefixes: ipPrefixes}
+	}
+
+	if outIps := idsToResourceReferences(config["outbound_ip_address_ids"]); outIps != nil {
+		outboundIps = &containerservice.ManagedClusterLoadBalancerProfileOutboundIPs{PublicIPs: outIps}
+	}
+
+	return &containerservice.ManagedClusterLoadBalancerProfile{
+		ManagedOutboundIPs: managedOutboundIps,
+		OutboundIPPrefixes: outboundIpPrefixes,
+		OutboundIPs:        outboundIps,
+	}, nil
+}
+
+func idsToResourceReferences(set interface{}) *[]containerservice.ResourceReference {
+	if set == nil {
+		return nil
+	}
+
+	s := set.(*schema.Set)
+	results := make([]containerservice.ResourceReference, 0)
+
+	for _, element := range s.List() {
+		id := element.(string)
+		results = append(results, containerservice.ResourceReference{ID: &id})
+	}
+
+	if len(results) > 0 {
+		return &results
+	}
+
+	return nil
+}
+
+func resourceReferencesToIds(refs *[]containerservice.ResourceReference) []string {
+	if refs != nil {
```

Comment:
Could we invert this check
```suggestion
	if refs == nil { return nil } 
```